exporter/ingest/spreadsheet/handler.py

```diff
@@ -0,0 +1,31 @@
+from kombu import Message
+
+from exporter.ingest.service import IngestService
+from exporter.ingest.spreadsheet.message import SpreadsheetGeneratorMessage
+from exporter.queue.handler import MessageHandler
+from exporter.session_context import SessionContext
+
+
+class SpreadsheetHandler(MessageHandler):
+    def __init__(self, ingest_service: IngestService):
+        super().__init__('IngestSpreadsheetGenerator')
+        self.ingest = ingest_service
+
+    def set_context(self, body: dict) -> SessionContext:
+        return SessionContext(
+            logger=self.logger,
+            context={
+                'submission_uuid': body.get('submissionUuid'),
+                'export_job_id': body.get('exportJobId'),
+                'project_uuid': body.get('projectUuid')
+            }
+        )
+
+    def handle_message(self, body: dict, msg: Message):
+        message = SpreadsheetGeneratorMessage(body)
+        self.logger.info('Received spreadsheet generation message')
+        gen_job_id = self.ingest.start_spreadsheet_generation(message.submission_uuid)
```

Comment:
Rather than calling broker we should:
1. push the export code into client (its already mostly there as data collector)
2. Have the exporter run a version of [the aborted broker code](https://github.com/ebi-ait/ingest-broker/pull/44) for generating metadata
3. have the exporter trigger the new client spreadsheet exporter

Reply:
Are you saying that since the spreadsheet generation is a function in ingest client, then we might as well use it directly in the exporter instead of invoking the endpoint in the broker? That's a good idea. Less moving parts.

Reply:
pretty much.
For step 1: we have to move [this method](https://github.com/ebi-ait/ingest-broker/blob/5b16e79fb6eade97e7c446a9e14fdb67273fb11b/broker/submissions/export_to_spreadsheet_service.py#L84) from broker into ingest_client:
``` export_to_spreadsheet_service.py
    def export(self, submission_uuid: str):
        self.logger.info(f'Generating Spreadsheet: {submission_uuid}')
        entity_dict = self.data_collector.collect_data_by_submission_uuid(submission_uuid)
        entity_list = list(entity_dict.values())
        flattened_json = self.downloader.convert_json(entity_list)
        workbook = self.downloader.create_workbook(flattened_json)
        return workbook
```